### ISSUE TITLE
Docker Builds fail when being executed in parallel

### ISSUE THREAD

**Issue by mdegel** [conan-io/conan-package-tools#139]
When 2 or more Docker Builds are being executed at the same time, the first one will run as expected, however the others will fail with the following error:
```
docker: Error response from daemon: Conflict. The container name "/conan_runner" is already in use by container "[...]". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
sha256:[...]
Error response from daemon: You cannot remove a running container [...]. Stop the container before attempting removal or force remove
```
A common situation where this behaviour could occur, would be when a CI-Server starts multiple Jobs in parallel (e.g. when building multiple libs with the same docker image).

From what I can tell, the error seems to be caused by the run-method from the DockerTestPackageRunner, which seems to update the docker image (I think?): https://github.com/conan-io/conan-package-tools/blob/a577cb65ae486c0f8330d2b1f09d9698e17eb9e3/conan/create_runner.py#L127
If this part is indeed the issue, would it be an acceptable solution to provide another environment variable (e.g. DOCKER_IMAGE_UPGRADE=[True,False]) in order to check whether or not to do the pull step? This would also be useful, when working with preconfigured/prebuilt docker images which use specific conan versions.

**Comment by uilianries** [comment]
AFAIK conan-package-tools uses the same name for every container. So, the possible solution could be append a random string to each new container name.

**Comment by Fearxpc** [comment]
When you have a jenkins-slave which can run 5 jobs parallel and every job uses the same docker image - then

- DockerTestPackageRunner.run() would trigger to download the same image 5 times
- Starting 5 containers (if we add a random string to the docker container name)
- install/upgrade conan_package_tools and conan in each container
- commit each container layer to the same docker image and delete the random container

In our environment we already provide a docker image which includes conan_package_tools and conan and so we only need the pull image part of the run method

Therefore it would be nice to have a env var which controls if this "init" part is executed on the docker image or not

**Comment by lasote** [comment]
Try it in 0.8.4 and reopen the issue if it's not solved!

**Comment by mdegel** [comment]
Just tried the newest release, works as intended now for as much as I can see. Thanks a lot.
